### PR TITLE
Only support libffi on unix for now

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,24 +20,25 @@ doctest = false # and no doc tests
...
 getrandom = { version = "0.2", features = ["std"] }
 env_logger = "0.9"
-libffi = "3.0.0"
-libloading = "0.7"
 log = "0.4"
 shell-escape = "0.1.4"
 rand = "0.8"
 smallvec = "1.7"
 
+
```
Comment (saethlin) [inline Cargo.toml:28]: ?

File: Cargo.toml [hunk 2]
```diff
@@ -20,24 +20,25 @@ doctest = false # and no doc tests
...
 
-# Enable some feature flags that dev-dependencies need but dependencies
-# do not.  This makes `./miri install` after `./miri build` faster.
 [target."cfg(unix)".dependencies]
 libc = "0.2"
+libffi = "3.0.0"
+libloading = "0.7"
 
+# Enable some feature flags that dev-dependencies need but dependencies
+# do not.  This makes `./miri install` after `./miri build` faster.
```
Comment (RalfJung) [inline Cargo.toml:41]: I think this comment is outdated, it refers to some feature flags we added on regular dependencies that were only needed for some dev-deps.

File: src/machine.rs [hunk 3]
```diff
@@ -410,6 +411,7 @@ pub struct MiriMachine<'mir, 'tcx> {
 
 impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
     pub(crate) fn new(config: &MiriConfig, layout_cx: LayoutCx<'tcx, TyCtxt<'tcx>>) -> Self {
+        #[cfg(unix)]
         let target_triple = &layout_cx.tcx.sess.opts.target_triple.to_string();
```
Comment (RalfJung) [inline src/machine.rs:415]:
```suggestion
```

File: src/machine.rs [hunk 4]
```diff
@@ -462,6 +464,7 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
             report_progress: config.report_progress,
             basic_block_count: 0,
             clock: Clock::new(config.isolated_op == IsolatedOp::Allow),
+            #[cfg(unix)]
             external_so_lib: config.external_so_file.as_ref().map(|lib_file_path| {
                 // Check if host target == the session target.
```
Comment (RalfJung) [inline src/machine.rs:469]:
```suggestion
                let target_triple = &layout_cx.tcx.sess.opts.target_triple.to_string();
                // Check if host target == the session target.
```

**Review by RalfJung — COMMENTED** [review]
r=me with the nits fixed

**Comment by RalfJung** [comment]
Oh also the readme should say that that flag is unix-only.

**Pull request merged**